lec04-dplyr.Rmd

```diff
@@ -1,4 +1,724 @@
+    group_by(year, species) %>%
+    tally()
+
+ggplot(yearly_counts, aes(x = n)) +
+    geom_histogram()
+```
+
+Creating an intermediate variable, would be prefereable if you are preferring a time consuming calculation, because you would not want to do that operation everytime you change some plot aesthetic.
+
+If it is not a time consuming approach and you would like the flexibility of
+changing the data summary and the plotting options in the same code chunk, you
+can pipe the output of your split-apply-combine operation to the plotting
+command:
+
+```{r}
+surveys_abun_species %>%
+    group_by(year, species) %>%
+    tally() %>% 
+    ggplot(aes(x = n)) +
+        geom_histogram()
+```
+
+We can perform a quick check that the plot corresponds to the table by coloring the histogram by species:
+
+```{r}
+surveys_abun_species %>%
+    group_by(year, species) %>%
+    tally() %>% 
+    ggplot(aes(x = n, fill = species)) + # `fill` is specific for histograms
+        geom_histogram()
+```
+
+
+Let's explore how the number of each genus varies over time. Longitudinal data
+can be visualized as a line plot with years on the x axis and counts on the y
+axis:
+
+```{r first-time-series}
+surveys_abun_species %>%
+    group_by(year, species) %>%
+    tally() %>% 
+    ggplot(aes(x = year, y = n)) +
+        geom_line()
+```
+
+Unfortunately, this does not work because we plotted data for all the species
+together. We need to tell ggplot to draw a line for each species by modifying
+the aesthetic function to include `group = species_id`:
+
+```{r time-series-by-species}
+surveys_abun_species %>%
+    group_by(year, species) %>%
+    tally() %>% 
+    ggplot(aes(x = year, y = n, group = species)) +
+        geom_line()
+```
+
+We will be able to distinguish species in the plot if we add colors (using `color` also automatically groups the data):
+
+```{r time-series-with-colors}
+surveys_abun_species %>%
+    group_by(year, species) %>%
+    tally() %>% 
+    ggplot(aes(x = year, y = n, color = species)) + # `color` groups automatically
+        geom_line() # try adding `size = 2`
+```
+
+### Faceting
+
+ggplot has a special technique called *faceting* that allows the user to split one plot
+into multiple subplots based on a variable included in the dataset. We will use it to
+make a time series plot for each species:
+
+```{r first-facet}
+surveys_abun_species %>%
+    group_by(year, species) %>%
+    tally() %>% 
+    ggplot(aes(x = year, y = n)) + #
+        geom_line() +
+        facet_wrap(~ species)
+```
+
+Now we would like to split the line in each plot by the sex of each individual
+measured. To do that we need to make counts in the data frame grouped by `year`,
+`species_id`, and `sex`:
+
+```{r}
+surveys_abun_species %>%
+    group_by(year, species, sex) %>%
+    tally()
+```
+
+We can make the faceted plot by splitting further by sex using `color` (within a single plot):
+
+```{r facet-by-species-and-sex}
```

Comment:
just curious. Any particular reason why you are naming some code chunks and not others?

Reply:
Because they were named in the data carpentry lecture this time, so I kept those names since it was convenient for troubleshooting. I will remove them, especially since I found the little box in the status bar that indicates with chunk number I am on.